examples/style.rs

```diff
@@ -0,0 +1,238 @@
+use std::{
+    error::Error,
+    io::{self, Stdout},
+    iter::once,
+    result,
+    time::Duration,
+};
+
+use crossterm::{
+    event::{self, Event, KeyCode},
+    execute,
+    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
+};
+use ratatui::{prelude::*, widgets::*};
+
+type Result<T> = result::Result<T, Box<dyn Error>>;
+
+fn main() -> Result<()> {
+    let mut terminal = setup_terminal()?;
+    let res = run_app(&mut terminal);
+    restore_terminal(terminal)?;
+    if let Err(err) = res {
+        eprintln!("This example will fail terminal height less than 32 lines.");
+        eprintln!("{err:?}");
+    }
+    Ok(())
+}
+
+fn run_app<B: Backend>(terminal: &mut Terminal<B>) -> io::Result<()> {
+    loop {
+        terminal.draw(ui)?;
+
+        if event::poll(Duration::from_millis(250))? {
+            if let Event::Key(key) = event::read()? {
+                if let KeyCode::Char('q') = key.code {
+                    return Ok(());
+                }
+            }
+        }
+    }
+}
+
+fn ui<B: Backend>(frame: &mut Frame<B>) {
+    let normal_colors = [
+        Color::Black,
+        Color::Red,
+        Color::Green,
+        Color::Yellow,
+        Color::Blue,
+        Color::Magenta,
+        Color::Cyan,
+        Color::Gray,
+    ];
+    let bright_colors = [
+        Color::DarkGray,
+        Color::LightRed,
+        Color::LightGreen,
+        Color::LightYellow,
+        Color::LightBlue,
+        Color::LightMagenta,
+        Color::LightCyan,
+        Color::White,
+    ];
+    render_bg_colors(frame, normal_colors, bright_colors);
+    render_fg_colors(frame, normal_colors, bright_colors);
+    render_indexed_colors(frame);
+    render_indexed_grayscale(frame);
+    render_modifiers(frame);
+}
+
+fn render_bg_colors<B: Backend>(
+    frame: &mut Frame<B>,
+    normal_colors: [Color; 8],
+    bright_colors: [Color; 8],
+) {
```

Comment:
I think it's already good and I'm not sure how much work you want to put into this to make it more easier for users to read. Some minor thoughts I had were it might be nicer if it was written like this:

```rust
fn render_bg_colors<B: Backend>(
    frame: &mut Frame<B>,
    normal_colors: [Color; 8],
    bright_colors: [Color; 8],
) {
    render_paragraphs(frame, 0, normal_colors, Color::White, "normal bg with white fg");
    render_paragraphs(frame, 1, bright_colors, Color::White, "bright bg with white fg");
    render_paragraphs(frame, 2, normal_colors, Color::Black, "normal bg with black fg");
    render_paragraphs(frame, 3, bright_colors, Color::Black, "bright bg with white fg");
}
```

If I were a complete noob and see the example screenshot and then look at this code, I imagine this might be much more clear as to what is going on?

Reply:
I ended up making a slew of changes to take some inspiration from https://gist.github.com/HaleTom/89ffe32783f89f403bba96bd7bcd1263, but that makes a whole tonne of vhs bug workaround that make the code even less readable :/

this is a good idea though.

Reply:
I did something similar to this.